### ISSUE TITLE
Bug when passing undefined parameter in place of callback

### ISSUE THREAD

**Issue by ORESoftware** [Automattic/mongoose#5061]
I have this:

```js
function create(data, cb){
  return Prompt.create(data, cb);
}
```

If callback is undefined, **I would expect it to return a Promise**, instead, it's doing some really weird thing, where the first parameter is no longer recognized, and instead it's looking at the second parameter which is of course undefined, and tries to use the second parameter to instantiate the object which of course will fail validation.

**Comment by varunjayaraman** [comment]
@ORESoftware that's because in the node world, a callback takes 2 parameters: the error and the result. You should omit the callback parameter if you want to use promises. 

For example:

```javascript
return Prompt
  .create(data)
  .then(result => doStuffWithResult(result))
  .catch(error => handleError(error));
```

**Comment by ORESoftware** [comment]
Right this relates to the other issue I opened, but this specifically targets the bug I was seeing when passing a null/undefined value, like so Model.create(data, null);

You're right, I can do this:

```js
function create(data, cb){
if(cb){
  return Prompt.create(data, cb);
}
else{
   return Prompt.create(data);
}
}
```

but isn't it obvious that the probably actually lies with the Mongoose code, not mine? I shouldn't have to do the above check. Mongoose should be handling it. Is this not clear?

**Comment by varunjayaraman** [comment]
Fair enough, we can add a check to see if `cb` is undefined.

**Comment by ORESoftware** [comment]
Yeah I can confirm this is a bug, I cannot figure out exactly why it's happening, but somehow I think you know what I am talking about.

The following screenshot shows that doc is defined, and that it has the fields "kind" and "value", yet I will get a validation error.

<img width="744" alt="screenshot 2017-03-10 11 45 19" src="https://cloud.githubusercontent.com/assets/11139560/23810736/a003e948-0587-11e7-929b-bd1114362b37.png">

**Comment by ORESoftware** [comment]
Just to summarize, when my code looks like this:

```js
function create(data, cb){
  if(cb){
    Prompt.create(data, cb)
  }
  else{
    return Prompt.create(data);
  }
}
```

then it works

but when the code is like this:

```js
function create(data, cb){
  return Prompt.create(data,cb);
}

```

then it will fail, with bizarre results.

that is why I **know** there is a bug in the Mongoose library


If you need more information please let me know. thanks.

**Comment by vkarpov15** [comment]
Thanks for reporting, will be fixed with 4.8.7 :+1:
